e2e/app/page/workspace-base.ts

```diff
@@ -94,5 +97,48 @@ export default abstract class WorkspaceBase extends AuthenticatedPage {
     return waitForAttributeEquality(this.page, {xpath: selector}, 'aria-selected', 'true');
   }
 
+  /**
+   * Delete ConceptSet, Dataset or Cohort thru Ellipsis menu located inside the resource card.
+   * @param {string} resourceName
+   * @param {ResourceCard} resourceType
+   */
+  async deleteResource(resourceName: string, resourceType: ResourceCard): Promise<string[]> {
+    const resourceCard = new DataResourceCard(this.page);
+    const card = await resourceCard.findCard(resourceName, resourceType);
+    if (!card) {
+      throw new Error(`Failed to find ${resourceType} card "${resourceName}"`);
+    }
+
+    await card.clickEllipsisAction(EllipsisMenuAction.Delete, {waitForNav: false});
+
+    const modal = new Modal(this.page);
+    await modal.waitForLoad();
+    const modalTextContent = await modal.getTextContent();
+
+    let link;
+    switch (resourceType) {
+    case ResourceCard.Cohort:
+      link = LinkText.DeleteCohort;
+      break;
+    case ResourceCard.ConceptSet:
+      link = LinkText.DeleteConceptSet;
```

Comment:
Question: If we delete a concept Set or Cohort that is being used in existing dataSet ,there will be another Modal with warning text , is that covered?

Reply:
Currently it's covered. I created a new ticket to track work. [RW-5589](https://precisionmedicineinitiative.atlassian.net/browse/RW-5589)